From a6ca11ddde28df631218b1465c8778df122fbb9b Mon Sep 17 00:00:00 2001
From: Tom Wieczorek <twieczorek@mirantis.com>
Date: Mon, 15 Sep 2025 15:10:48 +0200
Subject: [PATCH 3/5] Fix compile errors around integer conversions on 32-bit

Signed-off-by: Tom Wieczorek <twieczorek@mirantis.com>
---
 felix/config/param_types.go | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/felix/config/param_types.go b/felix/config/param_types.go
index 4a33a98134..a47cca28d4 100644
--- a/felix/config/param_types.go
+++ b/felix/config/param_types.go
@@ -196,9 +196,9 @@ func intSchema(ranges []MinMax) string {
 }
 
 func formatInt(m int) string {
-	if m == math.MaxInt64 {
+	if int64(m) == math.MaxInt64 {
 		return "2^63-1"
-	} else if m == math.MinInt64 {
+	} else if int64(m) == math.MinInt64 {
 		return "-2^63"
 	}
 	return fmt.Sprint(m)
@@ -900,7 +900,7 @@ func (r *RegionParam) SchemaDescription() string {
 
 // linux can support route-table indices up to 0xFFFFFFFF
 // however, using 0xFFFFFFFF tables would require too much computation, so the total number of designated tables is capped at 0xFFFF
-const routeTableMaxLinux = 0xffffffff
+const routeTableMaxLinux uint32 = 0xffffffff
 const routeTableRangeMaxTables = 0xffff
 
 type RouteTableRangeParam struct {
-- 
2.50.1

